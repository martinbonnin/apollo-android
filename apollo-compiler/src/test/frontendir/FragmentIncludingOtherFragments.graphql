query GetAnimal {
  ...queryFragment
}

fragment queryFragment on Query {
  randomAnimal {
    ...catFragment
    ...lionFragment
  }
}

fragment catFragment on Cat {
  name
  habitat {
    temperature
  }
}
fragment lionFragment on Lion {
  name
}