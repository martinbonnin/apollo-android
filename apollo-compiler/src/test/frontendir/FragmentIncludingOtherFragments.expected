package com.example

import kotlin.Float
import kotlin.String

interface GetAnimal {
  interface Query {
    val randomAnimal: Animal?

    interface AnimalCat {
      val name: String

      val habitat: com.example.Habitat

      interface Habitat {
        val temperature: Float?
      }
    }

    interface AnimalLion {
      val name: String
    }
  }
}

interface QueryFragment {
  interface Query {
    val randomAnimal: Animal?

    interface AnimalCat {
      val name: String

      val habitat: com.example.Habitat

      interface Habitat {
        val temperature: Float?
      }
    }

    interface AnimalLion {
      val name: String
    }
  }
}

interface CatFragment {
  interface Cat {
    val name: String

    val habitat: com.example.Habitat

    interface Habitat {
      val temperature: Float?
    }
  }
}

interface LionFragment {
  interface Lion {
    val name: String
  }
}
